src/api/tape.jl

```diff
@@ -212,6 +212,8 @@ end
 # JacobianTape #
 ################
 
+LinearAlgebra.lu(x::Adjoint, args...) = LinearAlgebra.lu(Array(x), args...)
```

Comment:
would it be possible to definite a `lu(x::TrackedArray{Adjoint,..}, ...)` instead here to avoid type piracy?

Reply:
I'm not sure, since this is so unrelated to the other changes to me it seems generally a bit weird. Why do we have to "fix" `lu` if the problem is the accumulation of the derivatives? I understand (or assume) it is necessary to fix some test errors but I think this points to another deeper problem or at least requires a more general solution.

Reply:
Yeah, the introduction of `adjoint` in DiffRules also causes failure of another test case which is unrelative to broadcasting on Adjoint:
https://github.com/JuliaDiff/ReverseDiff.jl/blob/master/test/api/JacobianTests.jl#L293

My intention was indeed to fix it, but I am not familiar with LA and Jacobian Matrix, I just went through the error log and found it needs such a method. Sorry about that. I hope somebody would like to dig deeper into this. Thanks.

Reply:
No worries :slightly_smiling_face: I just found https://github.com/JuliaDiff/ReverseDiff.jl/issues/183#issuecomment-920026895, so there's definitely a lot of not nice/non-general code regarding `Adjoint` already in ReverseDiff :see_no_evil: Would make it even nicer to fix these problems more generally.

Reply:
Since this is functionally correct, shall we 1) open a new issue 2) leave more improvements as separate PR?